src/metametric/core/matching.py

```diff
@@ -22,7 +22,11 @@ def __init__(self, inner: Metric[T], constraint: Union[str, MatchingConstraint]
 
     def score(self, x: Collection[T], y: Collection[T]) -> float:
         """Score two sets of objects."""
-        if isinstance(self.inner, DiscreteMetric) and self.constraint == MatchingConstraint.ONE_TO_ONE:
+        if not x and not y:
```

Comment:
Do you mean that `x is not None`? If so, could you make it explicit? Otherwise, there might be undefined behavior for certain type.

Reply:
No, I’m interested in checking for the emptiness of the collections. So I think this should be fine?